src/pegasus/response.lua

```diff
@@ -213,13 +214,58 @@ function Response:write(body, stayOpen)
   return self
 end
 
-function Response:writeFile(file, contentType)
-  self:contentType(contentType)
-  self:statusCode(200)
-  local value = file:read('*a')
+local function readfile(filename)
+  local file, err = io.open(filename, 'rb')
+  if not file then
+    return nil, err
+  end
+
+  local value, err = file:read('*a')
   file:close()
-  self:write(value)
+  return value, err
+end
+
+-- return nil+err if not ok
+function Response:writeFile(filename, contentType)
+  if type(filename) ~= "string" then
+    -- deprecated backward compatibility; file is a file-descriptor
+    self:contentType(contentType)
+    self:statusCode(200)
+    local value = filename:read('*a')
+    filename:close()
+    self:write(value)
+    return self
+  end
+
+  local contents, err = readfile(filename)
+  if not contents then
+    return nil, err
+  end
+
+  self:statusCode(200)
+  self:contentType(contentType or mimetypes.guess(filename))
+  self:write(contents)
+  return self
+end
+
+-- download by browser, return nil+err if not ok
+function Response:sendFile(path)
+  local filename = path:match("[^/]*$") -- only filename, no path
+  self:addHeader('Content-Disposition', 'attachment; filename="' .. filename .. '"')
+
+  local ok, err = self:writeFile(path, 'application/octet-stream')
+  if not ok then
```

Comment:
```suggestion
  if err then
```

Reply:
checking on `ok` is idiomatic in Lua. Only if an expected, positive, result value is either `nil` or `false` the check on `err` is used.